tools/load-generator/main.go

```diff
@@ -120,16 +132,59 @@ func NewQuerier(groupID int, target, prNumber string, qg QueryGroup) *Querier {
 	}
 }
 
-func (q *Querier) run(wg *sync.WaitGroup) {
+// Function to load `minTime` and `maxTime` from bucket-config.yml
+func loadKeyConfig() (*BucketConfig, error) {
+	filePath := "/config/bucket-config.yml"
+	_, err := os.Stat(filePath)
+	if os.IsNotExist(err) {
+		return nil, fmt.Errorf("file not found: %s", filePath)
+	}
+
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file: %w", err)
+	}
+
+	var bucketConfig BucketConfig
+	err = yaml.Unmarshal(data, &bucketConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing YAML: %w", err)
+	}
+
+	return &bucketConfig, nil
+}
+
+func configstate(v *BucketConfig, err error) *configState {
+	var absolutetime int64
+	if v != nil {
+		absolutetime = v.MaxTime
+	}
+	return &configState{
+		bucketConfig: v,
+		Err:          err,
```

Comment:
I think the idea is we carry `err` around, and act like this config doesn't exist if `err` is non-nil.

This at least deserves a comment.  Maybe an alternative would work, just having `nil` for this `configState` object, if there would be an error, and checking that?